### ISSUE TITLE
Inefficiency Decoder Training with Multi-GPU on Several Nodes (Machines)

### ISSUE THREAD

**Issue by 1073521013** [lucidrains/DALLE2-pytorch#156]
Hello everyone, training with multi-GPU on one node (machine) is normal, but multi-GPU on several nodes (machines) training is Inefficiency. By sorting out the problem, it is found that it has nothing to do with communication, storage, etc. Compared with single machine, the time is consumed in `self.accelerator.backward(loss)`. DDP should use autograd hooks registered at construction to trigger gradient synchronization and wait for all-reduce to complete. Have you encountered similar problems?

**Comment by nousr** [comment]
> The time is consumed in self.accelerator.backward(loss)

I would imagine that multi-node training would take a performance hit compared to the same # of GPUs on a single machine as you need to communicate gradients both ways across some kind of network. 

From what I understand, you are claiming this performance hit is more than expected? If you have a comparison with pure PyTorch DDP, I'm sure this would be valuable information for the huggingface team.

**Comment by Veldrovive** [comment]
Backward is where I would expect to see the performance hit since it is doing a gather with a very large set of gradients.
